### PR TITLE
fix: get ts declarations working when imported into a ts project and reduce bundle size

### PR DESCRIPTION
**Pull request by alittlebitweird**
* bundle declarations to one `index.d.ts` file at dist root
* mark dependencies as external so rollup doesn't package them all (bundle was 1.7mb lol, now it's 85kb)
* add analyzer rollup plugin to visualize bundle size

### CODE REVIEW

**Comment by alittlebitweird** [comment]
@brycedev i started a new `example-template-ts` repo to serve as a base for future client-specific repos and realized that the ts definitions weren't working 😅 

went to peep the explore tab on the [npm page](https://www.npmjs.com/package/@editframe/editframe-js) to try to figure out why and saw this error lol

Screen Shot 2022-03-08 at 1 53 20 AM
 
 So i ran a bundle analyzer and saw that a) the bundle size was like 1.5mb 💀
 
 long story short, now its down to ~85kb 😄 and I got the declaration files bundling down to a single `index.d.ts` file at the root of the dist, and it seems to be working 💪🏻

**Review by brycedev — APPROVED** [review]
😅 oof. thanks for catching this

**Comment by brycedev** [comment]
we like 85kb
